ocaml/typing/typemod.ml

```diff
@@ -3686,10 +3692,19 @@ let report_error ~loc _env = function
       Location.errorf ~loc
         "@[Top-level module bindings must have layout value, but@ \
          %s has layout@ %a.@]" id Sort.format sort
+ | Strengthening_mismatch(lid, explanation) ->
+      let main = Includemod_errorprinter.err_msgs explanation in
+      Location.errorf ~loc
+        "@[<v>\
+           @[In this strengthened module type, the type of %a@ \
+             does not match the underlying type@]@ \
+           %t@]"
+        longident lid main
   | Cannot_implement_parameter path ->
       Location.errorf ~loc
         "@[Interface %s@ found for this unit is flagged as a parameter.@ \
-         It cannot be implemented directly. Use -as-argument-for instead.@]"
+         It cannot be implemented directly. Recompile it without@ \
+         -as-parameter.@]"
```

Comment:
I think it would be better not to give a suggestion here. It's hard to guess what the user was actually trying to do, and the error is already pretty clear.